lib/floe/workflow.rb

```diff
@@ -161,13 +161,20 @@ def current_state
     private
 
     def step_next
-      if context.next_state
-        # if rerunning due to an error (and we are using Retry)
-        if context.state_name == context.next_state && context.failed? && context.state.key?("Retrier")
-          restore_values = context.state.slice("RetryCount", "Input", "Retrier")
-        end
-        context.state = {"Name" => context.next_state, "Input" => context.output}.merge!(restore_values || {})
+      context.state_history << context.state
```

Comment:
Nicely shifts the ownership of history to the workflow

Reply:
adam mentioned in the chat room:

This will set the state_history for all but the last state.
I'm suggesting moving it to the tail end of step_nonblock like we did in #206